sim-lib/src/lib.rs

```diff
@@ -1,44 +1,87 @@
 use bitcoin::secp256k1::PublicKey;
 use lightning::ln::PaymentHash;
-use serde::{Deserialize, Serialize};
-use std::time::SystemTime;
+use serde::{de::Visitor, Deserialize, Deserializer, Serialize, Serializer};
+use std::{collections::HashMap, str::FromStr, sync::Arc, time::SystemTime};
+
+pub mod lnd;
 
 // Phase 0: User input - see config.json
 
 // Phase 1: Parsed User Input
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct NodeConnection {
-    id: String,
-    host: String,
-    macaroon: String,
-    cert: String,
+    pub id: String,
+    pub address: String,
+    pub macaroon: String,
+    pub cert: String,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct Config {
-    nodes: Vec<NodeConnection>,
+    pub nodes: Vec<NodeConnection>,
+    pub activity: Vec<ActivityDefinition>,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ActivityDefinition {
     // The source of the action.
-    source: PublicKey,
+    #[serde(
+        serialize_with = "serialize_public_key",
+        deserialize_with = "deserialize_public_key"
+    )]
+    pub source: PublicKey,
     // The destination of the action.
-    dest: PublicKey,
+    #[serde(
+        serialize_with = "serialize_public_key",
+        deserialize_with = "deserialize_public_key"
+    )]
+    pub dest: PublicKey,
     // The frequency of the action, as in number of times per minute.
```

Comment:
I introduced this in a previous PR, but it looks like our frequency is not expressed as times per minutes but as delay between actions. e.g. a frequency of 5 would imply a payment every 5 seconds.